interview-questions.md

```diff
@@ -5,55 +5,54 @@ First, without external resources. Challenge yourself to answer from memory.
+1. What is a block in Ruby? 
 
-  Your answer:
-
-  Researched answer:
+  Your answer: A block in Ruby is an anonymous function and can be created with do/end keywords.
 
+  Researched answer: Ruby has an anonymous function similar to Javascript that are called blocks. Ruby blocks are codes that can be passed into methods. To create a block is necessary to close using do/end statements or with curly brackets {}. Blocks also have arguments that should be defined between two pipe | characters.
 
 
 2. What is a gem?
 
-  Your answer:
+  Your answer: I do not remember what acgem is. I will need to research this answer. 
 
-  Researched answer:
+  Researched answer: A gem is a sofware package in which Ruby applications or libraries can be distributed in a single format.
 
 
 
 3. What is Ruby on Rails?
 
-  Your answer:
+  Your answer: Ruby on Rails is a server-side web application framework written in Ruby. 
 
-  Researched answer:
+  Researched answer: Ruby on Rails is an open-source sofware used to build web applications. Rails was created by Danish programmer David Heinemeier Hansson and released in July 2004 as an open source code. Ruby on Rails follows an MVC architectural pattern. Architectural patterns are a way of dividing up responsibilities within the application. MVC stands for Model, View, and Controller and provide default structures for a database, a web service, and web pages. 
 
 
 
-4. What is a relational database? Are there other kinds of databases?
+4. What is a relational database? Are there other kinds of databases? 
 
-  Your answer:
+  Your answer: A relational database is like an Excel spreadsheet with tables containing rows and columns. That are many other kind of databases like Oracle, MySQL, etc. 
 
-  Researched answer:
+  Researched answer: A relational datase is a collection of information that organizes data for easier access. SQR is a domain-specific language used for communicating with relational databases. A database can have many tables that have relationships to each other. The description of the layout of all the tables in a database and the name and data type of each column within each table in a database is called a schema. There are other kinds of databases like object-oriented database systems, network database systems, and hierarchical database systems.
 
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: Primary key are unique identifier. They are important because they connect databases. We can use it to access a particular method. 
 
-  Researched answer:
+  Researched answer: Primary keys are important because they offer a unique identifier for a row of database tables. It also makes it easier for the user to add, sort, modify or even delete data in a database. A primary key has an essential role in database programming, as it ensures that each record is distinct from other records, and it is responsible for helping not to create duplicate data.
```

Comment:
Great answers in this section!  Especially your researched answers are thorough and well put.